### PR TITLE
First draft of delay distributions and biases session

### LINKED ISSUE #18: Practice session: Simulating biases in delay distributions and estimating delays without adjustment on these data
As the title make a first draft along the lines of #10

### PR DESCRIPTION
**Pull request by sbfnk**
initial skeleton for a delay distributions session. work in progress

Closes #18

### CODE REVIEW

File: Rmd/delay_distributions.Rmd
Comment (seabbs) [inline Rmd/delay_distributions.Rmd:122]: This is better than above but isn't "right" so do we want to touch on that?

Comment (sbfnk) [reply Rmd/delay_distributions.Rmd:122]: yes and the address in full estimation below with stan model I think

File: scripts/snippets/inf-hosp.r
Comment (seabbs) [inline scripts/snippets/inf-hosp.r:6]: if start_date is a date then this data won'
t have double censoring (as the input is discrete).

File: delay_distributions.Rmd
Comment (seabbs) [inline delay_distributions.Rmd:42]: This seems way to hard to me? Unless we are doing a lot of work in the lecture to teach what the delay process is

File: scripts/snippets/inf-hosp.r
Comment (seabbs) [reply scripts/snippets/inf-hosp.r:6]: (this is why we used a continuous model to simulate infections in the delay dist long paper)

File: scripts/snippets/inf-hosp.r
Comment (seabbs) [inline scripts/snippets/inf-hosp.r:7]: If infection time discrete? It needs to be continuous for double censoring to be present

File: scripts/snippets/inf-hosp.r
Comment (seabbs) [inline scripts/snippets/inf-hosp.r:9]: Is it going to be clear that `as.Date` acts as a flooring mechanism?

File: scripts/snippets/inf-hosp.r
Comment (seabbs) [inline scripts/snippets/inf-hosp.r:11]: Do you reeally need this complexity for the lesson at hand?

File: delay_distributions.Rmd
Comment (seabbs) [inline delay_distributions.Rmd:99]: An option here: https://mc-stan.org/cmdstanr/articles/r-markdown.html

File: delay_distributions.Rmd
Comment (seabbs) [inline delay_distributions.Rmd:144]: I thought we discussed this and agree it was best to stick to the solution that is closest to best practice and mirrors the simulation?

File: delay_distributions.Rmd
Comment (seabbs) [inline delay_distributions.Rmd:198]: Is the lecture going to cover some of the why as this just feels pulled out of thin air?

**Review by seabbs — COMMENTED** [review]
Just had another look. Not clear where this is at so comments may be a bit confused.

File: delay_distributions.Rmd
Comment (sbfnk) [reply delay_distributions.Rmd:99]: That's nice. It won't work if people prefer to copy & paste.

File: delay_distributions.Rmd
Comment (sbfnk) [reply delay_distributions.Rmd:42]: What would be the right level of difficulty in your view?

File: delay_distributions.Rmd
Comment (sbfnk) [reply delay_distributions.Rmd]: Fixed in 46477ce

File: scripts/snippets/inf-hosp.r
Comment (sbfnk) [reply scripts/snippets/inf-hosp.r:7]: it's continuous

File: scripts/snippets/inf-hosp.r
Comment (sbfnk) [reply scripts/snippets/inf-hosp.r:9]: Possibly not. I'm not a big fan of this either. Do you have a suggestion for a clearer way of doing this (alternatively can add an explanation).

File: scripts/snippets/inf-hosp.r
Comment (sbfnk) [reply scripts/snippets/inf-hosp.r:11]: Perhaps not - although it helps illustrate the point that doing so leads to a less sharp time distribution as in the figure that follows. That's a point I think we should try to get across. Do you have a suggestion for a better way of doing so?

File: delay_distributions.Rmd
Comment (sbfnk) [reply delay_distributions.Rmd:144]: Would be happy to have a solution with latent infection times but when I did so the stan code turned quite complex. I was probably missing a trick.

File: delay_distributions.Rmd
Comment (sbfnk) [reply delay_distributions.Rmd:198]: I think ideally this session would be followed by an explanation why, i.e. do first then see the theory behind it.

**Comment by sbfnk** [comment]
> Just had another look. Not clear where this is at so comments may be a bit confused.

Thanks! Definitely work in progress but very useful to have the approach challenged at this stage.

File: delay_distributions.Rmd
Comment (seabbs) [reply delay_distributions.Rmd:198]: I see

File: scripts/snippets/inf-hosp.r
Comment (sbfnk) [reply scripts/snippets/inf-hosp.r]: Turns out it actually doesn't, and this was news to me. We may have to use `lubridate` or one way or the other definitely something clearer.

``` r
date <- as.Date("2024-05-24")
fractional <- as.Date(date + 0.2)
fractional
#> [1] "2024-05-24"
as.POSIXct(fractional)
#> [1] "2024-05-24 04:48:00 UTC"
```

<sup>Created on 2024-05-24 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

File: delay_distributions.Rmd
Comment (sbfnk) [reply delay_distributions.Rmd]: One way to simplify this would be to not work with dates but instead numeric days (starting with 0). We could then introduce dates in #54.

File: scripts/snippets/inf-hosp.r
Comment (sbfnk) [reply scripts/snippets/inf-hosp.r]: see https://github.com/nfidd/nfidd/pull/17#discussion_r1624199126

File: delay_distributions.Rmd
Comment (sbfnk) [reply delay_distributions.Rmd]: Suggestion from conversation outside this PR: error on 0-1 delays, which simplifies things

**Comment by sbfnk** [comment]
Updated session to now address censoring and truncation separately.

The truncation model currently does not yield great estimates, though I haven't found an error yet.

File: stan/censored_delay_model.stan
Comment (seabbs) [inline stan/censored_delay_model.stan:24]: Using a beta prior makes this not quite a traditional censoring model (a rare case where uniform is maybe okay) but should only make a small difference.

File: stan/censored_delay_model.stan
Comment (seabbs) [inline stan/censored_delay_model.stan:22]: These priors are very wide. I can see this throwing some issues potentially?

File: stan/truncated_delay_model.stan
Comment (seabbs) [inline stan/truncated_delay_model.stan:17]: Hmm is this equivalent to dividing by the CDF? I think it isn't but I would be hard pressed to give a ref to that effect. Is there are model problems thought this is where I would expect them to be coming from.

Comment (seabbs) [reply stan/truncated_delay_model.stan:17]: Reading this again would certainly seem to imply that this works: https://mc-stan.org/docs/stan-users-guide/truncation-censoring.html#truncated-data.section

**Comment by seabbs** [comment]
This looks really good. 

Its a shame not to cover both at once but I can see why. If time I suggest we code up a mode that shows this. 

> The truncation model currently does not yield great estimates, though I haven't found an error yet.

Seem my query as to if the automated notation is properly normalising. I don't know but perhaps its worth trying. I think merge this and promote that + prior exploration to an issue.

File: stan/censored_delay_model.stan
Comment (sbfnk) [reply stan/censored_delay_model.stan:22]: Making them narrower might actually solve the issue in the truncated model (as it's finding some parameter combinations that are so right-heavy that the denominator gets shrunk).

File: stan/truncated_delay_model.stan
Comment (sbfnk) [reply stan/truncated_delay_model.stan:17]: When I tried both they seemed to give the same results.

**Comment by sbfnk** [comment]
> Its a shame not to cover both at once but I can see why. If time I suggest we code up a mode that shows this.
> 
> >

Yes - also a good exercise to explore for those who rush through the session.

**Comment by sbfnk** [comment]
> I think merge this and promote that + prior exploration to an issue.

Just waiting for your approval for merge.

**Review by seabbs — APPROVED** [review]

**Pull request merged**
